From f1c9af6e602c5d34767113cb4d2f1970584a7fd1 Mon Sep 17 00:00:00 2001
From: Ryan QIAN <b32804@freescale.com>
Date: Mon, 23 Jul 2012 15:57:52 +0800
Subject: [PATCH 1799/2508] ENGR00217936-01 mmc: esdhc: fix unknown controller
 version for usdhc

SVN value (0x3) defined in fsl host controller on mx6dq/mx6dl differs from
the one (0x2) defined in sdhc specification.

- original 0x11 is an incorrect value, it should be 0x3

Note: mx6sl has no such issue because it's fixed by IC, in other word,
mx6sl aligns with sdhc specification.

Signed-off-by: Ryan QIAN <b32804@freescale.com>
---
 drivers/mmc/host/sdhci-esdhc-imx.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/mmc/host/sdhci-esdhc-imx.c b/drivers/mmc/host/sdhci-esdhc-imx.c
index 30ec9f7..1111185 100644
--- a/drivers/mmc/host/sdhci-esdhc-imx.c
+++ b/drivers/mmc/host/sdhci-esdhc-imx.c
@@ -70,7 +70,8 @@
 #define SDHCI_PROT_CTRL_1BIT		(0 << 1)
 #define SDHCI_PROT_CTRL_LCTL		(1 << 0)
 
-#define SDHCI_FSL_SVN_300			0x11
+/* SDHCI_FSL_SVN_300 only for mx6q and mx6dl */
+#define SDHCI_FSL_SVN_300			0x3
 
 #define SDHCI_TUNE_CTRL		0xCC
 #define SDHCI_TUNE_CTRL_STD_TUNING_EN		(1 << 24)
-- 
1.7.7.4

